IBAnimatable/BackgroundImageDesignable.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Created by phimage on 25/03/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+
+import UIKit
+
+/// Protocol for designing background image
+public protocol BackgroundImageDesignable {
+
+  /**
+   * The background image
+   */
+  var backgroundImage: UIImage? { get set }
+
+}
+
+/// Protocol for designing background
+public protocol BackgroundDesignable: class {
+
+  /**
+   * The background view
+   */
+  var backgroundView: UIView? { get set }
+
+}
+
+extension UITableViewCell: BackgroundDesignable {}
+extension UITableViewHeaderFooterView: BackgroundDesignable {}
+extension UITableView: BackgroundDesignable {}
+extension UICollectionViewCell: BackgroundDesignable {}
+extension UICollectionView: BackgroundDesignable {}
```

Comment:
It took me a while to understand this, I found out is a great idea for having a `protocol BackgroundDesignable` Because we can't use protocol extension for conforming different classes like `public extension BackgroundImageDesignable where Self: UITableViewCell or UITableViewHeaderFooterView or UITableView or UICollectionViewCell or UICollectionView`. To have `BackgroundDesignable` and make the code looks better. Well done 👍